src/components/geo/render.js

```diff
@@ -71,25 +71,29 @@ export const renderMap = ({ref, width, height, resultsData, modelViewSelected, s
       resultsData.counts.forEach((d) => {demeValues[d.key] = d[vizCategory];});
       domainMax = resultsData.maxValue;
     } else {
-      resultsData.percentages.forEach((d) => {demeValues[d.key] = d[vizCategory];});
+      resultsData.mapData.forEach((d) => {demeValues[d.key] = d[vizCategory];});
       domainMax = 100;
     }
 
-    if (!modelViewSelected) {
+    if (categories.length === 1) {
```

Comment:
This feels fragile to me. What if another categorical variable happens to only have one level?

Reply:
I agree. I think the data-flow here needs an overhaul in general.